### PR TITLE
Make `Scattering1D` compatible with TF>=2.5

### PR DESCRIPTION
**Pull request by lostanlen**
as discussed in #775
done in duet with @MuawizChaudhary 
ready for code review
i still need to update some docs
triaged for v0.3

once this PR passes code review, i'll work on 2D and 3D

### CODE REVIEW

**Comment by lostanlen** [comment]
not sure how i'm supposed to test that? create a new CI job?

File: examples/1d/plot_filters.py [hunk 1]
```diff
@@ -90,11 +90,11 @@ def scattering1D_filter_factory():
   # explanatory labels and title.
   
   plt.figure()
-  plt.plot(np.arange(N)/N, phi_f[0], 'r')
   
+  plt.plot(np.arange(N)/N, phi_f['levels'][0], 'r')
   for psi_f in psi1_f:
-      plt.plot(np.arange(N)/N, psi_f[0], 'b')
   
+      plt.plot(np.arange(N)/N, psi_f['levels'][0], 'b')
```
Comment (MuawizChaudhary) [inline examples/1d/plot_filters.py:97]: We can fix this later, but I dont see why this had to change lines.

**Review by MuawizChaudhary — APPROVED** [review]
I am immediately merging based on the fact that it fixes #775 and #801 (as seen in [this notebook](https://colab.research.google.com/drive/1xXgDsAc5uPMEn7jtYUsAvsksigdTgSlb?usp=sharing).)  @janden you are welcome to look over the diff and make comments on it yourself, but given the severity of this issue impacting an entire frontend, I'm immediately merging. We can fix anything in a future PR. 

Great work @lostanlen !

File: tests/scattering1d/test_torch_scattering1d.py [hunk 2]
```diff
@@ -135,10 +135,8 @@ def test_computation_Ux(backend, device, random_state=42):
         for k in range(len(scattering.psi1_f)):
             assert (k,) in s.keys()
         for k in s.keys():
-            if k is not ():
+            if k != ():
                 assert k[0] < len(scattering.psi1_f)
-            else:
-                assert True
```
[the remark this answers is not in the document]
Comment (lostanlen) [reply tests/scattering1d/test_torch_scattering1d.py:141]:
```
> k is not ()
<stdin>:1: SyntaxWarning: "is not" with a literal. Did you mean "!="?
```

[the remark this answers is not in the document]
Comment (lostanlen) [reply tests/scattering1d/test_torch_scattering1d.py:141]: (initially spotted by @changhongw 🙏🏻 )

**Comment by lostanlen** [comment]
2D, you're next

**Pull request merged**